impl/http/src/main/java/io/serverlessworkflow/impl/executors/http/auth/requestbuilder/AccessTokenProvider.java

```diff
@@ -15,38 +15,46 @@
  */
 package io.serverlessworkflow.impl.executors.http.auth.requestbuilder;
 
+import static io.serverlessworkflow.api.types.OAuth2TokenRequest.Oauth2TokenRequestEncoding.APPLICATION_X_WWW_FORM_URLENCODED;
+
 import io.serverlessworkflow.impl.TaskContext;
+import io.serverlessworkflow.impl.WorkflowContext;
+import io.serverlessworkflow.impl.WorkflowError;
+import io.serverlessworkflow.impl.WorkflowException;
+import io.serverlessworkflow.impl.WorkflowModel;
+import io.serverlessworkflow.impl.executors.http.HttpClientResolver;
 import io.serverlessworkflow.impl.executors.http.auth.jwt.JWT;
 import io.serverlessworkflow.impl.executors.http.auth.jwt.JWTConverter;
+import jakarta.ws.rs.ProcessingException;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.client.Invocation;
+import jakarta.ws.rs.client.ResponseProcessingException;
+import jakarta.ws.rs.client.WebTarget;
+import jakarta.ws.rs.core.Form;
+import jakarta.ws.rs.core.GenericType;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.ServiceLoader;
 
 public class AccessTokenProvider {
```

Comment:
Can we replace this provider in the extension? For this feature, we will be using the Quarkus OIDC Client, which we will likely be using.

Reply:
The Provider suffix here is kind of misleading, but I keep the original because lack of better alternative (Im terrible with naming). This is class is just an aggregation of the info that is needed to perform the call, using JaxRX, to retrieve the token. 
If in future we use Quarkus OIDC client, probably it will be better to write a new HTTP task  (which can be done, base on the quarkus stack, in the quarkus extension)